auth_jwt/models/auth_jwt_validator.py

```diff
@@ -170,17 +175,48 @@ def _get_and_check_uid(self, payload):
             raise InternalServerError()
         return uid
 
+    def _create_partner_from_payload(self, payload):
+        return self.env["res.partner"].create(
+            {
+                "name": payload.get("name", "Guest"),
+                "phone": payload.get("phone_number"),
+                "email": payload.get("email"),
+                "website": payload.get("website"),
+                "auth_jwt_email": payload.get("email"),
+                "created_by_jwt": True,
+            }
+        )
+
+    def _get_partner_from_email(self, email):
+        partner = self.env["res.partner"].search([("auth_jwt_email", "=", email)])
+
+        if not len(partner):
+            partner = self.env["res.partner"].search(
+                [("email", "=", email), ("auth_jwt_email", "=", False)]
+            )
+
+        return partner
+
     def _get_partner_id(self, payload):
         # override for additional strategies
-        if self.partner_id_strategy == "email":
+        if self.partner_id_strategy in ["email", "email_create"]:
```

Comment:
Could you implement the new strategy in a separate branch? I'm not too comfortable with the mixing of the two strategies, as it makes the code harder to read. And since this is security sensitive, it's even more important to keep the code as simple as possible.